index.js

```diff
@@ -2,19 +2,18 @@
 
 /* eslint-disable no-console, prefer-template */
 
-// The official way to annotate overloaded functions: https://github.com/facebook/flow/issues/60
-declare function printNodeType(node: *): string;
-// eslint-disable-next-line no-redeclare
-declare function printNodeType(node: *, branch: string): void;
-
-function printTree(
-  initialTree: *,
-  printNode: printNodeType,
-  getChildren: (node: *) => Array<*>,
-): void {
-  function printBranch(tree: *, branch: *): void {
+type PrintNode<T> = (node: T, branch: ?string) => ?string;
+type GetChildren<T> = (node: T) => Array<T>;
```

Comment:
Using this approach we get type coverage when defining the `printNode` and `getChildren` functions